java/src/com/ibm/streamsx/rest/StreamsConnection.java

```diff
@@ -190,64 +33,13 @@ String getResponseString(String inputString) throws IOException {
      * @return a single {@link Instance}
      * @throws IOException
      */
-    public Instance getInstance(String instanceId) throws IOException {
-        Instance si = null;
-        if (instanceId.equals("")) {
-            // should add some fallback code to see if there's only one instance
-            throw new IllegalArgumentException("Missing instance name");
-        } else {
-            String instanceURL = url + "/instances/" + instanceId;
-            String sReturn = getResponseString(instanceURL);
-
-            si = Instance.create(this, sReturn);
-        }
-        return si;
-    }
-
-    /**
-     * This function is used to disable checking the trusted certificate chain
-     * and should never be used in production environments
-     * 
-     * @param allowInsecure
-     *            <ul>
-     *            <li>true - disables checking</li>
-     *            <li>false - enables checking (default)</li>
-     *            </ul>
-     * @return a boolean indicating the state of the connection after this
-     *         method was called.
-     *         <ul>
-     *         <li>true - if checking is disabled</li>
-     *         <li>false - if checking is enabled</li>
-     *         </ul>
-     */
-    public boolean allowInsecureHosts(boolean allowInsecure) {
-        try {
-            if ((allowInsecure) && (false == allowInsecureHosts)) {
-                CloseableHttpClient httpClient = HttpClients.custom()
-                        .setHostnameVerifier(new AllowAllHostnameVerifier())
-                        .setSslcontext(new SSLContextBuilder().loadTrustMaterial(null, new TrustStrategy() {
-                            public boolean isTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
-                                return true;
-                            }
-                        }).build()).build();
-                executor = Executor.newInstance(httpClient);
-                allowInsecureHosts = true;
-            } else if ((false == allowInsecure) && (true == allowInsecureHosts)) {
-                executor = Executor.newInstance();
-                allowInsecureHosts = false;
-            }
-        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
-            executor = Executor.newInstance();
-            allowInsecureHosts = false;
-        }
-        if (allowInsecureHosts) {
-            traceLog.info("Insecure Host Connection enabled");
-        }
-        return allowInsecureHosts;
-    }
+    Instance getInstance(String instanceId) throws IOException;
 
     /**
      * Cancels a job at this streams connection identified by the jobId
+     * FIXME: Does this really belong at this level or should it be in Instance
+     * or even Job? I think the current implementation only works on the default
+     * domain/instance
```

Comment:
+1 It doesn't make sense to have it here.

Reply:
I agree, I think this should be moved to `Job`.

Reply:
For binary and source compatibility I'll have to leave it in the `StreamsConnection` class, but I'll remove it from the new interface, and add it to `Job`.